src/metadata.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::HashMap;
+
+use crate::{
+    account::{account, Account},
+    amount::{amount, currency, Amount},
+    date::{date, Date},
+    string::string,
+};
+
+use nom::{
+    branch::alt,
+    bytes::complete::tag,
+    character::complete::{line_ending, one_of, satisfy, space0, space1},
+    combinator::{map, recognize},
+    multi::{fold_many0, many0},
+    sequence::{pair, preceded, separated_pair, tuple},
+    IResult,
+};
+
+pub type Metadata<'a> = HashMap<String, Value<'a>>;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Value<'a> {
```

Comment:
Mark it non-exhaustive, so that adding more variants won't be a breaking change.

```suggestion
#[non_exhaustive]
pub enum Value<'a> {
```